matrix/lambdas/daemons/driver.py

```diff
@@ -21,15 +24,20 @@ def run(self, request_id: str, bundle_fqids: typing.List[str], format: str):
         :param bundle_fqids: List of bundle fqids to be queried on
         :param format: MatrixFormat file format of expression matrices
         """
-        self.dynamo_handler.create_state_table_entry(request_id, len(bundle_fqids))
+        num_expected_mappers = math.ceil(len(bundle_fqids) / self._bundles_per_worker)
+        self.dynamo_handler.create_state_table_entry(request_id, num_expected_mappers)
         self.dynamo_handler.create_output_table_entry(request_id)
 
-        for fqid in bundle_fqids:
-            bundle_uuid, bundle_version = fqid.split(".", 1)
+        iter_args = [iter(bundle_fqids)] * self._bundles_per_worker
+        for bundle_fqid_group in itertools.zip_longest(*iter_args):
+            bundle_fqid_group = list(filter(None, bundle_fqid_group))
+
+            bundle_uuids = [fqid.split(".", 1)[0] for fqid in bundle_fqid_group]
+            bundle_versions = [fqid.split(".", 1)[1] for fqid in bundle_fqid_group]
```

Comment:
Think it makes more sense to pass in `bundle_fqids` and perform the splits in the distributed mappers. Not a big deal, just a preference

Reply:
Agree. Wasn't sure if you guys had a reason for splitting it up in the driver, but sounds like no, so I'll push it down a bit further.